homeassistant/components/proximity.py

```diff
@@ -0,0 +1,325 @@
+                are:
+                'not set'
+                'arrived'
+                'towards'
+                'away_from'
+                'unknown'
+                'stationary'
+dist_to_zone = distance from the monitored zone (in km)
+
+Use configuration.yaml to enable the user to easily tune a number of settings:
+- Zone: the zone to which this component is measuring the distance to. Default
+  is the home zone
+- Ignored Zones: where proximity is not calculated for a device (either the
+  device being monitored or ones being compared (e.g. work or school)
+- Devices: a list of devices to compare location against to check closeness to
+  the configured zone
+- Tolerance: the tolerance used to calculate the direction of travel in metres
+  (to filter out small GPS co-ordinate changes
+
+Logging levels debug, info and error are in use
+
+Example configuration.yaml entry:
+proximity:
+  zone: home
+  ignored_zones:
+    - twork
+    - elschool
+  devices:
+    - device_tracker.nwaring_nickmobile
+    - device_tracker.eleanorsiphone
+    - device_tracker.tsiphone
+  tolerance: 50
+"""
+
+import logging
+from homeassistant.helpers.event import track_state_change
+from homeassistant.helpers.entity import Entity
+from homeassistant.util.location import distance
+
+DEPENDENCIES = ['zone', 'device_tracker']
+
+# domain for the component
+DOMAIN = 'proximity'
+
+# default tolerance
+DEFAULT_TOLERANCE = 1
+
+# default zone
+DEFAULT_PROXIMITY_ZONE = 'home'
+
+# entity attributes
+ATTR_DIST_FROM = 'dist_to_zone'
+ATTR_DIR_OF_TRAVEL = 'dir_of_travel'
+ATTR_NEAREST = 'nearest'
+ATTR_FRIENDLY_NAME = 'friendly_name'
+
+# Shortcut for the logger
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup(hass, config):  # pylint: disable=too-many-locals,too-many-statements
+    """ get the zones and offsets from configuration.yaml"""
+    ignored_zones = []
+    if 'ignored_zones' in config[DOMAIN]:
+        for variable in config[DOMAIN]['ignored_zones']:
+            ignored_zones.append(variable)
+            _LOGGER.info('ignored zones loaded: %s', variable)
+
+    # get the devices from configuration.yaml
+    if 'devices' not in config[DOMAIN]:
+        _LOGGER.error('devices not found in config')
+        return False
+    else:
+        proximity_devices = []
+        for variable in config[DOMAIN]['devices']:
+            proximity_devices.append(variable)
+            _LOGGER.info('proximity device added: %s', variable)
+
+    # get the direction of travel tolerance from configuration.yaml
+    if 'tolerance' in config[DOMAIN]:
+        tolerance = config[DOMAIN]['tolerance']
+    else:
+        tolerance = DEFAULT_TOLERANCE
+    _LOGGER.debug('tolerance set to: %s', tolerance)
+
+    # get the zone to monitor proximity to from configuration.yaml
+    if 'zone' in config[DOMAIN]:
+        proximity_zone = config[DOMAIN]['zone']
+    else:
+        proximity_zone = DEFAULT_PROXIMITY_ZONE
+    _LOGGER.info('zone set to %s', proximity_zone)
```

Comment:
There are a lot of debug logs going on. Consolidate them at the end in 1 statement .